docs/mkdocs/labsv2.md

```diff
@@ -0,0 +1,425 @@
+
+To minimize the impact on the workload running in the cluster, kernel module upgrades should be performed on a per-node basis.
+
+First, label the specific node that requires the kernel module upgrade. Use the following format:
+
+```
+kmm.node.kubernetes.io/version-module.<module-namespace>.<module-name>=$moduleVersion
+```
+
+In our case we'll use this example:
+```bash
+oc label node my-node-1 kmm.node.kubernetes.io/version-module.labsv2.kmm-ci-a=0.9
+```
+
+Then we can use a new build based on previous build example by just adapting the ModuleLoader (adding `version` and specific `containerImage` for that version) and reusing existing Configmap for build. Make sure you have deleted any Module object left from past examples and apply following Module:
+
+`build_with_version.yaml`
+```yaml
+---
+apiVersion: kmm.sigs.x-k8s.io/v1beta1
+kind: Module
+metadata:
+  name: kmm-ci-a
+  namespace: labsv2
+spec:
+  moduleLoader:
+    serviceAccountName: labsv2sa
+    container:
+      version: "0.9"
+      modprobe:
+        moduleName: kmm-ci-a
+      kernelMappings:
+        - regexp: '^.*'
+          containerImage: myrepo.xyz/kmm-labs:kmm-kmod-${KERNEL_FULL_VERSION}-v0.9
+          build:
+            dockerfileConfigMap:
+              name: build-module-labs
+  imageRepoSecret: 
+    name: myrepo-pull-secret
+  selector:
+    node-role.kubernetes.io/worker: ""
+
+```
+
+A new module named `kmm-ci-a` with object version 0.9 and image `kmm-kmod-${KERNEL_FULL_VERSION}-v0.9` will be created at node `my-node-1`.
+If we look back at the first example in the Lab about In-Cluster building from sources we will find that containerImage was `kmm-kmod-${KERNEL_FULL_VERSION}-v1.0` and if made all the examples we should have that v1.1 image at our registry.
+Let's pretend that we want to update our actual v0.9 module to v1.0.
+
+1. Check in advance that you have both v0.9 and v1.0 images at your registry.
+2. Remove 0.9 label from the node:
+```
+oc label node my-node-1 kmm.node.kubernetes.io/version-module.labsv2.kmm-ci-a-
+```
+3. Check that unlabeling the node killed the Module pods.
+4. Label with the new desired version:
+```
+oc label node my-node-1 kmm.node.kubernetes.io/version-module.labsv2.kmm-ci-a=1.0
+```
+5. Edit the module object and change both `version` and `containerImage` with the new version number and matching image 
+   (In our example 1.0 for the version and `kmm-kmod-${KERNEL_FULL_VERSION}-v1.0` for the containerImage):
+```
+oc edit module kmm-ci-a
+```
+6. The new module version is loaded.
+
+
+## Loading soft dependencies
+
+Kernel modules sometimes have dependencies on others modules so those dependencies have to be loaded in advance. In the following example `kmm-ci-a` depends on `kmm-ci-b` so we will set `modulesLoadingOrder` and then the list with the `moduleName` as the first entry followed by all of its dependencies. In our case it is just `kmm-ci-b` but it could be a longer list with multiple dependency modules.
```

Comment:
we should add that soft dependency should be used when there is a logical dependency between modules,but no  symbols' dependency (otherwise modprobe  would have done the work automatically)